From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Michal Wajdeczko <michal.wajdeczko@intel.com>
Date: Fri, 14 Aug 2026 19:17:47 +0200
Subject: [PATCH] drm/xe/survivability: Report sysfs failure in one place

If we fail to create any of the survivability sysfs files, actual
error is reported by the caller. Move that reporting to sysfs helper.

Signed-off-by: Michal Wajdeczko <michal.wajdeczko@intel.com>
Cc: Rodrigo Vivi <rodrigo.vivi@intel.com>
Cc: Riana Tauro <riana.tauro@intel.com>
Cc: Aravind Iddamsetty <aravind.iddamsetty@intel.com>
Cc: Mallesh Koujalagi <mallesh.koujalagi@intel.com>
Reviewed-by: Rodrigo Vivi <rodrigo.vivi@intel.com>
Link: https://patch.msgid.link/20260814171801.13347-22-michal.wajdeczko@intel.com
Signed-off-by: Rodrigo Vivi <rodrigo.vivi@intel.com>
(backported from commit 33cc29e3dc0e6530a1eb0a318ff7a414c0c21068 linux-next)
Signed-off-by: Bommu Krishnaiah <krishnaiah.bommu@intel.com>
Assisted-by: GNAI:gpt-5.3-codex
Signed-off-by: Akanksha Hotta <akanksha.hotta@intel.com>
---
 drivers/gpu/drm/xe/xe_survivability_mode.c | 23 +++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/drivers/gpu/drm/xe/xe_survivability_mode.c b/drivers/gpu/drm/xe/xe_survivability_mode.c
index 0ace10ae500a..4fe44bdc0161 100644
--- a/drivers/gpu/drm/xe/xe_survivability_mode.c
+++ b/drivers/gpu/drm/xe/xe_survivability_mode.c
@@ -18,6 +18,7 @@
 #include "xe_mmio.h"
 #include "xe_nvm.h"
 #include "xe_pcode_api.h"
+#include "xe_printk.h"
 #include "xe_vsec.h"
 
 /**
@@ -118,7 +119,7 @@ static const char * const reg_map[] = {
 	[POSTCODE_TRACE_OVERFLOW] = "Postcode trace overflow",
 	[AUX_INFO0]               = "Auxiliary Info 0",
 	[AUX_INFO1]               = "Auxiliary Info 1",
-	[AUX_INFO2]               = "Auxiliary Info 2",
+[AUX_INFO2]               = "Auxiliary Info 2",
 	[AUX_INFO3]               = "Auxiliary Info 3",
 	[AUX_INFO4]               = "Auxiliary Info 4",
 };
@@ -309,24 +310,26 @@ static int create_survivability_sysfs(struct pci_dev *pdev)
 	int ret;
 
 	ret = device_create_file(dev, &dev_attr_survivability_mode);
-	if (ret) {
-		dev_warn(dev, "Failed to create survivability sysfs files\n");
-		return ret;
-	}
+	if (ret)
+		goto failed;
 
 	ret = devm_add_action_or_reset(xe->drm.dev,
 				       xe_survivability_mode_fini, xe);
 	if (ret)
-		return ret;
+		goto failed;
 
 	/* Survivability info is not required if enabled via configfs */
 	if (!xe_configfs_get_survivability_mode(pdev)) {
 		ret = devm_device_add_group(dev, &survivability_info_group);
 		if (ret)
-			return ret;
+			goto failed;
 	}
 
 	return 0;
+
+failed:
+	xe_err(xe, "Failed to create survivability sysfs files: %pe\n", ERR_PTR(ret));
+	return ret;
 }
 
 static int enable_boot_survivability_mode(struct pci_dev *pdev)
@@ -406,7 +409,7 @@ bool xe_survivability_mode_is_requested(struct xe_device *xe)
 	if (survivability_mode)
 		return true;
 
-	data = xe_mmio_read32(mmio, PCODE_SCRATCH(0));
+data = xe_mmio_read32(mmio, PCODE_SCRATCH(0));
 	survivability->boot_status = REG_FIELD_GET(BOOT_STATUS, data);
 
 	return check_boot_failure(xe);
@@ -433,8 +436,7 @@ void xe_survivability_mode_runtime_enable(struct xe_device *xe)
 
 	populate_survivability_info(xe);
 
-	if (create_survivability_sysfs(pdev))
-		dev_err(&pdev->dev, "Failed to create survivability sysfs\n");
+	create_survivability_sysfs(pdev);
 
 	survivability->type = XE_SURVIVABILITY_TYPE_RUNTIME;
 	dev_err(&pdev->dev, "Runtime Survivability mode enabled\n");
@@ -470,7 +472,6 @@ int xe_survivability_mode_boot_enable(struct xe_device *xe)
 		log_survivability_info(xe);
 		return -ENXIO;
 	}
-
 	survivability->type = XE_SURVIVABILITY_TYPE_BOOT;
 
 	return enable_boot_survivability_mode(pdev);
-- 
2.43.0

